### PR TITLE
Disable keyboard accessibility outside drawer and modal

### LINKED ISSUE #2690: Disable keyboard accessibility on Modal and Drawer background elements
**Description**
Similar to #2672 , need to disable keyboard TAB accessibility for elements behind the modal and drawer

**Expected result**
Background elements cannot be accessed through keyboard.

### PR DESCRIPTION
**Pull request by ndricimrr**
**Description**

Changes proposed in this pull request:

- Disables keyboard accessibility on all elements outside the drawer/modal when these are opened.

**Related issue(s)**

Resolves #2690

### CODE REVIEW

**Review by wdoberschuetz — APPROVED** [review]
LGTM.

File: core/src/Modal.svelte [hunk 1]
```diff
@@ -236,10 +236,13 @@
...
+    IframeHelpers.disableA11YKeyboardExceptClassName('.fd-dialog');
+    window.focus();
   });
 
   onDestroy(() => {
     EventListenerHelpers.removeEventListener('message', onMessage);
+    IframeHelpers.enableA11YKeyboardBackdropExceptClassName('.fd-dialog');
   });
 
   // [svelte-upgrade suggestion]
```
Comment (ndricimrr) [inline core/src/Modal.svelte:248]:
```suggestion
  //  [svelte-upgrade suggestion]
```

File: core/src/Modal.svelte [hunk 2]
```diff
@@ -236,13 +236,16 @@
 
   onMount(() => {
     EventListenerHelpers.addEventListener('message', onMessage);
+    IframeHelpers.disableA11YKeyboardExceptClassName('.fd-dialog');
```
Comment (ndricimrr) [inline core/src/Modal.svelte:239]: only on backdrop

File: core/src/Modal.svelte [hunk 3]
```diff
@@ -236,13 +236,18 @@
...
   onMount(() => {
     EventListenerHelpers.addEventListener('message', onMessage);
+    // only disable accessibility for all cases other than a drawer without backdrop
+    !(settings.isDrawer && !settings.backdrop) ? IframeHelpers.disableA11YKeyboardExceptClassName('.fd-dialog') : '';
+    window.focus();
   });
 
   onDestroy(() => {
     EventListenerHelpers.removeEventListener('message', onMessage);
+    // only disable accessibility for all cases other than a drawer without backdrop
```
Comment (ndricimrr) [inline core/src/Modal.svelte:246]:
```suggestion
    // only disable accessibility for all cases other than a drawer without backdrop
```

**Pull request merged**